### PR TITLE
Release 2.3 Updates: Fixes for contamination, UMI Handling, Template 2.11.1

### PR DESCRIPTION
**Pull request by apeltzer**
Related to #114 

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/smrnaseq/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/smrnaseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -64,23 +77,6 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
-|               | `--mirgenedb_mature`     |
-|               | `--mirgenedb_hairpin`    |
-|               | `--contamination_filter` |
-|               | `--rrna`                 |
-|               | `--trna`                 |
-|               | `--cdna`                 |
-|               | `--ncrna`                |
-|               | `--pirna`                |
-|               | `--other_contamination`  |
-
```
Comment (maxulysse) [inline CHANGELOG.md:83]: this was released, so CHANGELOG should not change in that part, no?

File: CHANGELOG.md [hunk 2]
```diff
@@ -98,20 +94,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
-| Old parameter        | New parameter    |
-| -------------------- | ---------------- |
-| `--conda`            | `--enable_conda` |
-| `--clusterOptions`   |                  |
-| `--publish_dir_mode` |                  |
-
-> **NB:** Parameter has been **updated** if both old and new parameter information is present.
-> **NB:** Parameter has been **added** if just the new parameter information is present.
-> **NB:** Parameter has been **removed** if parameter information isn't present.
-
```
Comment (maxulysse) [inline CHANGELOG.md:114]: this was released, so CHANGELOG should not change in that part, no?

File: CHANGELOG.md [hunk 3]
```diff
@@ -133,6 +118,7 @@ Note, since the pipeline is now using Nextflow DSL2, each process will be run wi
 | `seqkit`             | 0.16.0      | 2.0.0       |
 | `trim-galore`        | 0.6.6       | 0.6.7       |
 | `bioconvert`         | -           | 0.4.3       |
+| `umi_tools`          | -           | 1.1.2       |
```
Comment (maxulysse) [inline CHANGELOG.md:121]: was this just forgotten for that release?

**Comment by apeltzer** [comment]
@maxulysse I will add these bits back and also update what was updated in this PR later. Same as adding a proper testcase for the UMI workflow to make sure both the "old way" and "new way" of things work fine 👍🏻

File: conf/modules.config [hunk 4]
```diff
@@ -142,6 +151,90 @@ if (!params.skip_fastp) {
     }
 }
 
+if (params.with_umi && !params.skip_umi_extract) {
+    process {
+        withName: '.*:FASTQC_UMITOOLS_TRIMGALORE:UMITOOLS_EXTRACT' {
```
Comment (grst) [inline conf/modules.config:156]: shouldn't this be FASTQC_UMITOOLS_FASTP?

File: conf/modules.config [hunk 5]
```diff
@@ -130,6 +131,14 @@ if (!params.skip_fastp) {
...
         process {
+            withName: '.*:.*:FASTQC_FASTP:FASTQC_RAW' {
+                ext.args = '--quiet'
+                publishDir = [
+                    path: { "${params.outdir}/fastqc/raw" },
+                    mode: params.publish_dir_mode,
+                    saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
+                ]
+            }
             withName: '.*:.*:FASTQC_FASTP:FASTQC_TRIM' {
```
Comment (grst) [inline conf/modules.config:142]: shouldn't this be FASTQC_UMITOOLS_FASTP?

File: conf/test.config [hunk 6]
```diff
@@ -23,8 +23,8 @@ params {
 
     input            = 'https://github.com/nf-core/test-datasets/raw/smrnaseq/samplesheet/v2.0/samplesheet.csv'
     fasta            = 'https://github.com/nf-core/test-datasets/raw/smrnaseq/reference/genome.fa'
-    mature           = 'https://mirbase.org/download/CURRENT/mature.fa'
-    hairpin          = 'https://mirbase.org/download/CURRENT/hairpin.fa'
+    mature           = 'https://mirbase.org/download/mature.fa'
+    hairpin          = 'https://mirbase.org/download/hairpin.fa'
```
Comment (grst) [inline conf/test.config:27]: Could remove this altogether as it's inherited from `nextflow.config`?

File: modules/local/bowtie_map_contaminants.nf [hunk 7]
```diff
@@ -21,17 +21,20 @@ process BOWTIE_MAP_CONTAMINANTS {
...
+        --un ${meta.id}.${contaminant_type}.filter.unmapped.contaminant.fastq \\
         --very-sensitive-local \\
         -k 1 \\
-        -x \$INDEX \\
-        --un ${meta.id}.${contaminant_type}.filter.unmapped.contaminant.fastq \\
-        ${reads} \\
+        -S ${meta.id}.filter.contaminant.sam \\
         ${args} \\
-        -S ${meta.id}.filter.contaminant.sam > ${meta.id}.contaminant_bowtie.log 2>&1
+        > ${meta.id}.contaminant_bowtie.log 2>&1
```
Comment (grst) [inline modules/local/bowtie_map_contaminants.nf:37]: I don't think this is mentioned in the changelog?

File: subworkflows/local/umi_dedup.nf [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+        INDEX_GENOME ( fasta )
+        bt_index      = INDEX_GENOME.out.bowtie_indices
+        fasta_formatted = INDEX_GENOME.out.fasta
+        ch_versions     = ch_versions.mix(INDEX_GENOME.out.versions)
+    } else {
+        bt_index     = Channel.fromPath("${bt_index}**ebwt", checkIfExists: true).ifEmpty { exit 1, "Bowtie1 index directory not found: ${bt_index}" }
+        fasta_formatted = fasta
+    }
+
+    if (bt_index){
```
Comment (grst) [inline subworkflows/local/umi_dedup.nf:34]: this should always be true?

**Review by grst — CHANGES_REQUESTED** [review]
found a few minor issues, but LGTM otherwise

File: modules/local/bowtie_map_contaminants.nf (hunk 7 above)
Comment (apeltzer) [reply modules/local/bowtie_map_contaminants.nf:37]: It is now :)

**Comment by apeltzer** [comment]
@nf-core-bot pretty please fix linting

**Comment by apeltzer** [comment]
@nf-core-bot fix linting pretty please

File: nextflow.config [hunk 9]
```diff
@@ -233,6 +249,7 @@ plugins {
 
     id 'nf-validation' // Validation of pipeline parameters and creation of an input channel from a sample sheet
```
Comment (maxulysse) [inline nextflow.config:251]:
```suggestion
```

File: nextflow.config [hunk 10]
```diff
@@ -262,6 +279,9 @@ process.shell = ['/bin/bash', '-euo', 'pipefail']
 docker.registry = 'quay.io'
 podman.registry = 'quay.io'
```
Comment (maxulysse) [inline nextflow.config:280]:
```suggestion
apptainer.registry   = 'quay.io'
docker.registry      = 'quay.io'
podman.registry      = 'quay.io'
singularity.registry = 'quay.io'
```

File: subworkflows/local/mirna_quant.nf [hunk 11]
```diff
@@ -64,15 +57,28 @@ workflow MIRNA_QUANT {
...
+    ch_versions = ch_versions.mix(INDEX_HAIRPIN.out.versions)
+
+    BOWTIE_MAP_HAIRPIN ( reads_hairpin, hairpin_bowtie.collect() )
+    ch_versions = ch_versions.mix(BOWTIE_MAP_HAIRPIN.out.versions)
+
     BAM_STATS_HAIRPIN ( BOWTIE_MAP_HAIRPIN.out.bam, FORMAT_HAIRPIN.out.formatted_fasta )
     ch_versions = ch_versions.mix(BAM_STATS_HAIRPIN.out.versions)
 
+
+
```
Comment (maxulysse) [inline subworkflows/local/mirna_quant.nf:81]: why all the empty lines?

File: workflows/smrnaseq.nf [hunk 12]
```diff
@@ -64,13 +64,15 @@ if (!params.mirgenedb) {
...
-include { MIRDEEP2           } from '../subworkflows/local/mirdeep2'
+include { INPUT_CHECK           } from '../subworkflows/local/input_check'
+include { FASTQ_FASTQC_UMITOOLS_FASTP } from '../subworkflows/nf-core/fastq_fastqc_umitools_fastp'
+include { DEDUPLICATE_UMIS      } from '../subworkflows/local/umi_dedup'
+include { CONTAMINANT_FILTER    } from '../subworkflows/local/contaminant_filter'
+include { MIRNA_QUANT           } from '../subworkflows/local/mirna_quant'
+include { GENOME_QUANT          } from '../subworkflows/local/genome_quant'
+include { MIRTRACE              } from '../subworkflows/local/mirtrace'
+include { MIRDEEP2              } from '../subworkflows/local/mirdeep2'
+include { INDEX_GENOME          } from '../modules/local/bowtie_genome'
```
Comment (maxulysse) [inline workflows/smrnaseq.nf:75]:
```suggestion
include { INPUT_CHECK                 } from '../subworkflows/local/input_check'
include { FASTQ_FASTQC_UMITOOLS_FASTP } from '../subworkflows/nf-core/fastq_fastqc_umitools_fastp'
include { DEDUPLICATE_UMIS            } from '../subworkflows/local/umi_dedup'
include { CONTAMINANT_FILTER          } from '../subworkflows/local/contaminant_filter'
include { MIRNA_QUANT                 } from '../subworkflows/local/mirna_quant'
include { GENOME_QUANT                } from '../subworkflows/local/genome_quant'
include { MIRTRACE                    } from '../subworkflows/local/mirtrace'
include { MIRDEEP2                    } from '../subworkflows/local/mirdeep2'
include { INDEX_GENOME                } from '../modules/local/bowtie_genome'
```

**Review by maxulysse — APPROVED** [review]
minor changes

**Comment by grst** [comment]
@nf-core-bot fix linting

**Review by grst — APPROVED** [review]
LGTM now

**Pull request merged**